src/main/java/ch/njol/skript/classes/data/DefaultFunctions.java

```diff
@@ -711,6 +705,252 @@ public String[] executeSimple(Object[][] params) {
 						"\t\tset {_money} to formatNumber({money::%sender's uuid%})",
 						"\t\tsend \"Your balance: %{_money}%\" to sender")
 			.since("2.10");
+
+		Functions.registerFunction(new SimpleJavaFunction<Number>("mean", new Parameter[]{
+			new Parameter<>("numbers", DefaultClasses.NUMBER, false, null)
+		}, DefaultClasses.NUMBER, true) {
+			@Override
+			public Number @Nullable [] executeSimple(Object[][] params) {
+				Double total = 0d;
+				int length = params[0].length;
+				for (int i = 0; i < length; i++) {
+					Number number = (Number) params[0][i];
+					if (number instanceof Double doubleNumber && (doubleNumber.isInfinite() || doubleNumber.isNaN()))
+						return null;
+					if (total.isInfinite() || total.isNaN())
+						return null;
+					total += (number.doubleValue() - total) / (i + 1);
+				}
+				return new Number[]{total};
+			}
+		})
+			.description(
+				"Get the mean (average) of a list of numbers.",
+				"You cannot get the mean of a set of numbers that includes infinity or NaN."
+			)
+			.examples(
+				"mean(1, 2, 3) = 2",
+				"mean(0, 5, 10) = 5",
+				"mean(13, 97, 376, 709) = 298.75"
+			)
+			.since("INSERT VERSION");
+
+		Functions.registerFunction(new SimpleJavaFunction<Number>("median", new Parameter[]{
+			new Parameter<>("numbers", DefaultClasses.NUMBER, false, null)
+		}, DefaultClasses.NUMBER, true) {
+			@Override
+			public Number @Nullable [] executeSimple(Object[][] params) {
+				AtomicBoolean invalid = new AtomicBoolean(false);
+				// median requires the numbers to be sorted from lowest to biggest
+				Number[] sorted = Arrays.stream(params[0])
+					.filter(object -> {
+						if (!(object instanceof Number number))
+							return false;
+						if (number instanceof Double doubleNumber && doubleNumber.isNaN()) {
```

Comment:
I can't think of an exact case this happening but what if the number is a Float? Probably best to just call doubleValue and see if that's nan